### DISCUSSION TITLE
Configure Seq dashboards, signals, et al directly from Seq.json

### DISCUSSION THREAD

**Discussion by Aaronontheweb** [datalust/seq-tickets#1092] [Ideas]
I'm looking at hosting Seq inside Kubernetes and I wanted to see if it's possible to use a ConfigMap to bootstrap several default signals our development team would find useful when running our K8s cluster locally on their development machines. Is there an easy way for me to pre-populate some of the Signals in the workspace via Seq.json? 

I see I can export some of the Signals I've built to JSON via the UI, but I'm not sure from where I can consume them.

**Comment by nblumhardt** [comment]
Hey Aaron, thanks for dropping by :-)

Unfortunatley, no - though it's a feature we're actively exploring. In the next version, [the default admin user will be configurable from _Seq.json_](#791) (or through env vars), but we've stopped short of a full configuration experience this time around.

A customized "run" script that uses https://github.com/datalust/seqcli to import the JSON signals will work today, though. `seqcli` is shipped as a Docker image (`datalust/seqcli`) as well as zipped binaries, so it's not too difficult to maneuver into position.

```
cat signal.json | docker run --rm -it datalust/seqcli signal import -s http://localhost:5341
```

Let me know if this helps. I think we're about ready to leave this ticket open and pick this up again on the next round of planning.

**Comment by nblumhardt** [comment]
We're going to bundle `seqcli` in a future version of the docker image (most likely 2020.4-ish); with that done, we'll be able to support executing a "first run" shell script from the mounted data volume. Since the container will now accept default admin username and password, we can pre-configure `seqcli` and put it on the path so that the script can run `seqcli` commands to create/configure various aspects of the service.

**Comment by nblumhardt** [comment]
See also #982, which this feature should support/enable.

**Comment by Aaronontheweb** [comment]
That would be excellent
